src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -662,21 +666,43 @@ public void kill(Map<String, String> modelEnvVars) throws IOException, Interrupt
                 getListener().getLogger().println("Killing processes");
 
                 String cookie = modelEnvVars.get(COOKIE_VAR);
-
-                int exitCode = doLaunch(
+                int exitCode = 1;
+                if (this.isUnix()) {
+                    exitCode = doLaunch(
+                                    true,
+                                    null,
+                                    null,
+                                    null,
+                                    null,
+                                    "sh",
+                                    "-c",
+                                    "kill \\`grep -l '" + COOKIE_VAR + "=" + cookie
+                                            + "' /proc/*/environ | cut -d / -f 3 \\`")
+                            .join();
+                } else {
+                    try {
+                        FilePath mainScript = copyKillScript(workspace, "kill-processes-with-cookie", ".ps1");
+                        FilePath killProcessScript = copyKillScript(workspace, "kill-process-by-id", ".ps1");
+                        FilePath csCode = copyKillScript(workspace, "ProcessEnvironmentReader", ".cs");
+                        exitCode = doLaunch( // Will fail if the script is not present, but it was also failing before in all cases
                                 true,
                                 null,
                                 null,
                                 null,
                                 null,
-                                // TODO Windows
-                                "sh",
-                                "-c",
-                                "kill \\`grep -l '" + COOKIE_VAR + "=" + cookie
-                                        + "' /proc/*/environ | cut -d / -f 3 \\`")
-                        .join();
-
-                getListener().getLogger().println("kill finished with exit code " + exitCode);
+                                "powershell.exe", "-NoProfile", "-File",
+                                /* path to file may contain spaces so wrap in double quotes*/
+                                "\""+mainScript.getRemote()+"\"",
+                                "-cookie", cookie, "-csFile", "\""+csCode.getRemote()+"\"","-killScript", "\""+killProcessScript.getRemote()+"\""
```

Comment:
Is this formatting actually passing https://github.com/jenkinsci/kubernetes-plugin/blob/5054f32a4549db21d473028c32fb10fc80057578/pom.xml#L53 ?

Reply:
No, it wasn't :cry: